dashboard/app/models/course_version.rb

```
@@ -105,7 +99,8 @@ def self.add_course_version(course_offering, content_root)
       )
       course_version.published_state = content_root.published_state
     else
-      course_version = nil
+      # If content_root is not a UnitGroup, then we cannot create a CourseVersion.
+      return nil
```

it seems like we should be raising in this case. see below for what fails.  What do you think about opening a new work item to track this?
```
Dave-MBP:~/src/cdo2/dashboard (TEACH-1789/disallow-standalone-unit-creation *%)$ time bundle exec spring testunit test/models/course_version_test.rb test/models/sections/section_test.rb test/models/lesson_test.rb test/models/course_offering_test.rb

ERROR["test_add_course_offering_does_nothing_if_is_course_is_false_for_unit", "CourseOfferingTest", 13.65281599998707]
 test_add_course_offering_does_nothing_if_is_course_is_false_for_unit#CourseOfferingTest (13.65s)
Minitest::UnexpectedError:         RuntimeError: cannot create CourseVersion for content root "bogus-script-58" that is not a UnitGroup
            app/models/course_version.rb:103:in `add_course_version'
            app/models/course_offering.rb:106:in `add_course_offering'
            test/models/course_offering_test.rb:107:in `block in <class:CourseOfferingTest>'
            test/testing/setup_all_and_teardown_all.rb:36:in `run'

ERROR["test_concise_summarize:_section_with_a_coteacher", "SectionTest", 53.60864399996353]
 test_concise_summarize:_section_with_a_coteacher#SectionTest (53.61s)
Minitest::UnexpectedError:         RuntimeError: cannot create CourseVersion for content root "jigsaw" that is not a UnitGroup
            app/models/course_version.rb:103:in `add_course_version'
            app/models/course_offering.rb:106:in `add_course_offering'
            test/models/sections/section_test.rb:556:in `block in <class:SectionTest>'
            test/testing/setup_all_and_teardown_all.rb:36:in `run'

ERROR["test_summarize:_section_with_a_coteacher", "SectionTest", 59.32666999998037]
 test_summarize:_section_with_a_coteacher#SectionTest (59.33s)
Minitest::UnexpectedError:         RuntimeError: cannot create CourseVersion for content root "jigsaw" that is not a UnitGroup
            app/models/course_version.rb:103:in `add_course_version'
            app/models/course_offering.rb:106:in `add_course_offering'
            test/models/sections/section_test.rb:996:in `block in <class:SectionTest>'
            test/testing/setup_all_and_teardown_all.rb:36:in `run'

ERROR["test_add_course_version_does_nothing_for_script_without_CourseVersion_if_is_course_is_false", "CourseVersionTest", 61.79845800000476]
 test_add_course_version_does_nothing_for_script_without_CourseVersion_if_is_course_is_false#CourseVersionTest (61.80s)
Minitest::UnexpectedError:         RuntimeError: cannot create CourseVersion for content root "bogus-script-425" that is not a UnitGroup
            app/models/course_version.rb:103:in `add_course_version'
            test/models/course_version_test.rb:231:in `block in <class:CourseVersionTest>'
            test/testing/setup_all_and_teardown_all.rb:36:in `run'
```

---

I agree, thank you for pointing that out. Should this just be done now instead of opening a new ticket so that we don't have anything silently falling through the cracks in the meantime?

---

including in this PR works for me!